src/System.Management.Automation/engine/hostifaces/MshHostUserInterface.cs

```diff
@@ -390,7 +391,29 @@ internal void IgnoreCommand(string commandText, InvocationInfo invocation)
         /// so that when content is sent through Out-Default it doesn't
         /// make it to the actual host.
         /// </summary>
-        internal bool TranscribeOnly { get; set; }
+        internal bool TranscribeOnly => Interlocked.CompareExchange(ref _transcribeOnlyCount, 0, 0) != 0;
```

Comment:
This is pretty confusing for a getter. A getter shouldn't normally change the value (which this doesn't, but when I read `CompareExchange`, I assume it does and need to read the code more closely).

If you're worried about atomic reads - that's a non-issue, integers are always read atomically on any platform we'll target.

If you're worried about needing a memory barrier, I think you need volatile on the field and use `Thread.VolatileRead`.